bus-mapping/src/evm/opcodes/sload.rs

```diff
@@ -0,0 +1,182 @@
+        let stack_position = exec_step.stack().last_filled();
+
+        // Manage first stack read at latest stack position
+        gc_idx += 1;
+        exec_step.bus_mapping_instance_mut().push(container.insert(
+            StackOp::new(
+                RW::READ,
+                gc_idx.into(),
+                stack_position,
+                stack_value_read,
+            ),
+        ));
+
+        // Storage read
+        let storage_value_read =
+            *exec_step.storage().get_or_err(&stack_value_read)?;
+        gc_idx += 1;
+        exec_step.bus_mapping_instance_mut().push(container.insert(
+            StorageOp::new(
+                RW::READ,
+                gc_idx.into(),
+                storage_value_read,
+                storage_value_read,
+                stack_value_read,
+            ),
+        ));
+
+        // First stack write
+        gc_idx += 1;
+        exec_step.bus_mapping_instance_mut().push(container.insert(
+            StackOp::new(
+                RW::WRITE,
+                gc_idx.into(),
+                stack_position,
+                storage_value_read,
+            ),
+        ));
+
+        Ok(gc_idx - gc_idx_start)
+    }
+}
+
+#[cfg(test)]
+mod sload_tests {
+    use super::*;
+    use crate::{
+        evm::{
+            EvmWord, GasCost, GasInfo, GlobalCounter, Memory, OpcodeId,
+            ProgramCounter, Stack, StackAddress, Storage,
+        },
+        BlockConstants, ExecutionTrace,
+    };
+    use pasta_curves::pallas::Scalar;
+    use std::collections::HashMap;
+    use std::iter::FromIterator;
+
+    #[test]
+    fn sload_opcode_impl() -> Result<(), Error> {
+        let trace = r#"
+        [
+          {
+            "pc": 163,
+            "op": "SLOAD",
+            "gas": 5217,
+            "gasCost": 2100,
+            "depth": 1,
+            "stack": [
+              "0x0"
+            ],
+            "storage": {
+              "0000000000000000000000000000000000000000000000000000000000000000": "000000000000000000000000000000000000000000000000000000000000006f"
+            }
+          },
+          {
+            "pc": 97,
+            "op": "STOP",
+            "gas": 0,
+            "gasCost": 0,
+            "depth": 1,
+            "stack": [
+              "0x6f"
+            ]
+          }
+        ]
+        "#;
+
+        // Obtained trace computation
+        let obtained_exec_trace = ExecutionTrace::<Scalar>::from_trace_bytes(
+            trace.as_bytes(),
+            BlockConstants::default(),
+        )
+        .unwrap();
+
+        let mut container = OperationContainer::new();
+
+        // Generate Step1 corresponding to SLOAD
+        let mut step_1 = ExecutionStep {
+            memory: Memory::empty(),
+            stack: Stack(vec![EvmWord::from(0x0u32)]),
+            storage: Storage::new(HashMap::<_, _>::from_iter([(
+                EvmWord::from(0x0u32),
+                EvmWord::from(0x6fu32),
```

Comment:
Would probably be nice to add a method to encapsulate this code. Since seems to be the case that will be re-used quite a lot.